docs/development/extending.md

```diff
@@ -355,3 +355,486 @@ sorted_ip_list: "{{ ip_list | computacenter.utils.sort_ip }}"
+- The NAME should exactly match the name of the plugin everywhere else.
+- For details on the imports and base classes/helpers.
+
+Documentaion -> Declare option that are needed in the plugin.
+Examples -> Example with parameter for a inventory file to run the script.
+Python Code -> Different methods like verify_file, parse and more.
+
+```python
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+name: cisco_prime.py
+author:
+  - Kevin Blase
+  - Jonathan Schmidt
+short_description: Inventory source for Cisco Prime API.
+description:
+  - Builds inventory from Cisco Prime API.
+  - Requires a configuration file ending in C(prime.yml) or C(prime.yaml).
+    See the example section for more details.
+version_added: 1.0.0
+extends_documentation_fragment:
+  - ansible.builtin.constructed
+notes:
+  - Nothing
+options:
+  plugin:
+    description:
+      - The name of the Cisco Prime API Inventory Plugin.
+      - This should always be C(custom.inventory.cisco_prime).
+    required: true
+    type: str
+    choices: [ custom.inventory.cisco_prime ]
+...
+"""
+
+EXAMPLES = r"""
+---
+# Inventory File
+plugin: custom.inventory.cisco_prime
+api_user: "user123"
+api_pass: "password123"
+api_host_url: "host.domain.tld"
+"""
+
+import requests
+# import traceback
+# from ansible.errors import AnsibleParserError
+from ansible.inventory.group import to_safe_group_name
+from ansible.plugins.inventory import (
+    BaseInventoryPlugin,
+    Constructable,
+    to_safe_group_name,
+)
+
+class InventoryModule(BaseInventoryPlugin, Constructable):
+
+    NAME = 'custom.inventory.cisco_prime'  # used internally by Ansible, it should match the file name but not required
+
+    def verify_file(self, path):
+        valid = False
+        if super(InventoryModule, self).verify_file(path):
+            if path.endswith(('prime.yaml', 'prime.yml')):
+                valid = True
+            else:
+                self.display.vvv(
+                    'Skipping due to inventory source not ending in "prime.yaml" nor "prime.yml"')
+        return valid
+
+    def add_host(self, hostname, host_vars):
+        self.inventory.add_host(hostname, group='all')
+
+        for var_name, var_value in host_vars.items():
+            self.inventory.set_variable(hostname, var_name, var_value)
+
+        strict = self.get_option('strict')
+
+        # Add variables created by the user's Jinja2 expressions to the host
+        self._set_composite_vars(self.get_option('compose'), host_vars, hostname, strict=True)
+
+        # Create user-defined groups using variables and Jinja2 conditionals
+        self._add_host_to_composed_groups(self.get_option('groups'), host_vars, hostname, strict=strict)
+        self._add_host_to_keyed_groups(self.get_option('keyed_groups'), host_vars, hostname, strict=strict)
+...
+```
+
+The Python file needs to be stored in a collection, e.g.:
```

Comment:
Wenn hier richtigerweise auf Punkte aus dem Kapitel Extending Ansible verwiesen wird, dann bitte einen Link erstellen. Trotzdem gerne den Code-Block drin lassen, er gibt ja ein gutes Beispiel wie die Ordner-Struktur für Inventory Plugins aussehen muss.

Reply:
Hier ist mir nicht ganz klar was du meinst.

Reply:
War mit dem Hintergedanken dass die Beschreibung in das Kapitel *Inventory* verschoben wird, hat sich erledigt.